wol/main.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python3
+
+"""
+use wol to generate and send Wake-On-LAN (WOL) "Magic Packet" to specific interface
+
+Usage: wol_click [OPTIONS] INTERFACE TARGET_MAC
+
+  Generate and send Wake-On-LAN (WOL) "Magic Packet" to specific interface
+
+Options:
+  -b           Use broadcast MAC address instead of target device's MAC
+               address as Destination MAC Address in Ethernet Frame Header.
+               [default: False]
+  -p password  An optional 4 or 6 byte password, in ethernet hex format or
+               quad-dotted decimal  [default: ]
+  -c count     For each target MAC address, the count of magic packets to
+               send. count must between 1 and 5. This param must use with -i.
+               [default: 1]
+  -i interval  Wait interval milliseconds between sending each magic packet.
+               interval must between 0 and 2000. This param must use with -c.
+               [default: 0]
+  -v           Verbose output  [default: False]
+  -h, --help   Show this message and exit.
+
+Examples:
+  wol Ethernet10 00:11:22:33:44:55
+  wol Ethernet10 00:11:22:33:44:55 -b
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 00:22:44:66:88:aa
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 192.168.1.1 -c 3 -i 2000
+"""
+
+import binascii
+import click
+import copy
+import netifaces
+import os
+import socket
+import time
+
+CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
+EPILOG = """\b
+Examples:
+  wol Ethernet10 00:11:22:33:44:55
+  wol Ethernet10 00:11:22:33:44:55 -b
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 00:22:44:66:88:aa
+  wol Vlan1000 00:11:22:33:44:55,11:33:55:77:99:bb -p 192.168.1.1 -c 3 -i 2000
+"""
+ORDINAL_NUMBER = ["0", "1st", "2nd", "3rd", "4th", "5th"]
+ETHER_TYPE_WOL = b'\x08\x42'
+
+
+class MacAddress(object):
+    def __init__(self, mac):
+        self.mac = mac
+
+    def __str__(self):
+        return "%x:%x:%x:%x:%x:%x" % (self.mac[0], self.mac[1], self.mac[2], self.mac[3], self.mac[4], self.mac[5])
+
+    def __eq__(self, other):
+        return self.mac == other.mac
+
+    def to_bytes(self):
+        return copy.copy(self.mac)
+
+
+BROADCAST_MAC = MacAddress(b'\xff\xff\xff\xff\xff\xff')
+
+
+def is_root():
+    return os.geteuid() == 0
+
+
+def get_interface_operstate(interface):
+    with open('/sys/class/net/{}/operstate'.format(interface), 'r') as f:
+        return f.read().strip().lower()
+
+
+def get_interface_mac(interface):
+    mac = netifaces.ifaddresses(interface)[netifaces.AF_LINK][0].get('addr')
+    return MacAddress(binascii.unhexlify(mac.replace(':', '').replace('-', '')))
```

Comment:
There are many places you use this pattern:
```
binascii.unhexlify(mac.replace(':', '').replace('-', '')
```

can you extract a method for it?

Reply:
Thanks for pointing out. I've refactor the `MacAddress` class to integrate this logic.